### DISCUSSION TITLE
Unable to hide filter row when using tree data with `enableFiltering: false`

### DISCUSSION THREAD

**Discussion by hwlv** [ghiscoding/slickgrid-universal#1624] [Q&A] [answered]
I am using SlickGrid-Universal with tree data, and I want to hide the filter row at the top of the grid. I set `enableFiltering: false` in the configuration, but I received an error message stating: "[SlickGrid-Universal] It looks like you are trying to use Tree Data without using the filtering option."

**Comment by ghiscoding** [comment, marked as answer]
The message is correct, Tree Data requires and uses filtering to work, you cannot disable it. When you collapse a parent group, the grid is actually using Filter to filter out child rows, so that is why Filtering must be enabled. However, you could probably hide the filter row if you don't want to show filter, or add `filterable: false` to each column

**Comment by hwlv** [comment]
When setting filterable: false for each column, a blank placeholder row appears. Using CSS to hide the filter row, like .slickgrid-container .slick-headerrow { display: none; }, effectively hides the row but also creates unwanted blank space below the grid, affecting the layout.

Could you suggest an alternative approach to avoid the blank space issue? Any guidance on how to properly handle this situation would be greatly appreciated.
![image](https://github.com/user-attachments/assets/ee7bcb37-4f52-4172-a77f-066551c1c21a)

**Comment by ghiscoding** [comment]
Disabling all column filters will not remove the filter header row, you can use the `showHeaderRow: false` grid option and/or toggle it from the Grid Menu. Also, if you don't want to see the Grid Menu toggle filter command, you should also hide it from the menu `gridMenu: { hideToggleFilterCommand: true }`

All of these things should be in the documentations, [Grid Menu](https://ghiscoding.gitbook.io/slickgrid-universal/grid-functionalities/grid-menu). If the docs is missing this kind of info, then please contribute to the project by adding missing documentation. I don't want to answer all these questions every time, documentation should be the place to find all the answers to your questions

![image](https://github.com/user-attachments/assets/d18ea2ea-963b-4559-8f92-513fe67a0f90)

Also note that all of these error/warning messages that you see are there for a reason. If you want to understand why the message, you can ask but the fact is that all of these messages serve a purpose. If the message(s) aren't clear enough, then contribute a Pull Request to improve the message. 

EDIT 

I added more docs related to this discussion in commit 67dfac3

**Reply by hwlv** [reply]
Thanks again for your support and for pointing me in the right direction! It's very nice of you !
